### DISCUSSION TITLE
Deep Scrapping

### DISCUSSION THREAD

**Discussion by akishore06** [AutomaApp/automa#1792] [Q&A]
I have a list of records displayed on screen 50 records per page using pagination. Once I click on one record it will take me to another sub list with lot of records. The sublist also using pagination. Once I click on a record in sub list it will display company name, website, phone. How to scrape this data using Automa? Please help.
![requirement](https://github.com/user-attachments/assets/16cfb1d6-baec-4026-8ddf-d0b93f586875)

**Comment by jingbof** [comment]
Seems like you need to work with [Looping](https://docs.automa.site/workflow/looping.html#looping) + [Go Back Block](https://docs.automa.site/blocks/go-back.html#go-back-block) with a bit of JS code I guess

**Sample Workflow**
1. Go to URL (main list of records)
2. Loop (handle pagination for main list)
   - Click (on each record)
   - Wait (for sublist page to load)
3. Loop (handle pagination for sublist)
   - Click (on each sublist record)
   - Wait (for company detail page to load)
   - JS code block (extract company name, website, phone)
   - Go Back (or click back to the sublist)
4. Click (next page in sublist, if exists) -> repeat step 3
5. Click (next page in main list, if exists) -> repeat step 2
